src/OrchardCore/OrchardCore.Notifications.Core/NotificationConstants.cs

```diff
@@ -4,8 +4,11 @@ public class NotificationConstants
 {
     public const string NotificationCollection = "Notification";
 
-    /// <summary>
-    /// If this number is increased after the site is setup, you must alter the <see cref="Notification" /> using migration
-    /// </summary>
-    public const int NotificationIndexContentLength = 2500;
+    // Maximum length that MySql can support in an inner index under utf8mb4 collation is 768,
```

Comment:
Can't we store the 768 as part of the provider default settings so it is not hard coded here or other places?

Reply:
Yes, I thought about having for example a value depending on the provider. Hmm but potentially we can deploy data from a database to another, so I don't know, for now the issue/PR is just to notify the limitation but open to any suggestion.

Reply:
But 2500 chars is high for a column to include in a inner index, if only for searching we would need to remove duplicate terms, stop words, delimiters, like a search engine may do for indexing.

Just saw that you use your own `StripHTML()` method using a `Regex`, not sure it is good, we have a `RemoveTags()` string extension, also allowing to specify whether we `htmlDecode` or not.

Reply:
I agree that column isn't a good idea. It would be nice to at least have full-index search column. The fact we are searching large index is very slow when we have lots of indexes. Maybe we should consider using a search index for notification. The only downside would be that we would have to force using the search module.